### PR TITLE
Change Message Count Color

### PR DESCRIPTION
**Pull request by wgyt**
**Resolves**

image

**Changes**

Change the badge color to a dark orange

**Reason for changes**

Made so I can see it better

**Tests**

It works 
image

### CODE REVIEW

**Comment by retronbv** [comment]
darn it i wanted to find it so bad lmao

**Comment by retronbv** [comment]
how does it look in light mode?

**Comment by wgyt** [comment]
> how does it look in light mode?

still the same 
image

**Comment by retronbv** [comment]
> > how does it look in light mode?
> 
> still the same
> image

no, you dont understand..... how does it look in browser light mode???
image

**Comment by Endermaster3690** [comment]
what about the scratch color too?

**Comment by mxmou** [comment]
> image

This color is a bit ugly - could you use dark green instead?

**Comment by wgyt** [comment]
> snipped
> no, you dont understand..... how does it look in browser light mode???
> image

image


> > image
> 
> This color is a bit ugly - could you use dark green instead?

I could, should I use a color like the Github comment button?

**Comment by wgyt** [comment]
m

**Comment by RedGuy12** [comment]
why did you merge other prs into this one

make a new pr


p.s. maybe a setting on the addon to choose a color?

**Comment by WorldLanguages** [comment]
Note: this green was the same one used by Scratch Messaging Extension

**Comment by wgyt** [comment]
> why did you merge other prs into this one
> 
> make a new pr
> 
> p.s. maybe a setting on the addon to choose a color?

I didn't. I was just keeping this branch up to date with master
p.s. I guess I could? I probally won't tho.

**Comment by WorldLanguages** [comment]
PR is now ok, but why this color exactly?

**Comment by mxmou** [comment]
> PR is now ok, but why this color exactly?

Maybe `#008000` could be used instead, because it's already used to display the user's own username in comments.

**Comment by wgyt** [comment]
> PR is now ok, but why this color exactly?

I can read it.

> Maybe `#008000` could be used instead, because it's already used to display the user's own username in comments.

That's a good idea.

**Comment by TheColaber** [comment]
Feel like this could be a setting... in more settings... Tbh, I don't mind the green, but I understand if it's hard to read for others.

**Comment by wgyt** [comment]
> Feel like this could be a setting... in more settings... Tbh, I don't mind the current green, but I understand if it's hard to read for others.
> 
> ps... this branch is from the world-languages org??? lol

I should do that soon, but I'm in school right now.

ps... yes lol

**Comment by wgyt** [comment]
I just added the option to change the color, can someone test it?

**Comment by mxmou** [comment]
> if this is merged, can I be invited to the org?

See https://github.com/ScratchAddons/ScratchAddons/discussions/1183#discussioncomment-260417

**Comment by TheColaber** [comment]
With the current way you are doing this... if we have any more addons that have badges, they would also need that setting... is that how it should be done?

**Comment by wgyt** [comment]
> With the current way you are doing this... if we have any more addons that have badges, they would also need that setting... is that how it should be done?

I guess this could be a Scratch Addon wide setting, but no addons other then this one use it so 🤷🏻‍♂️

**Comment by wgyt** [comment]
> With the current way you are doing this... if we have any more addons that have badges, they would also need that setting... is that how it should be done?

https://github.com/ScratchAddons/ScratchAddons/issues/10 would need this to be made part of core

**Comment by RedGuy12** [comment]
I think that this we shouldn't need to bother about moving it to more settings until #10 is implemented.

**Comment by TheColaber** [comment]
This is ok as a temporary solution to just having one single addon using the badge. But I see 2 problems with this PR:
1. The default color should be the current, just so current users don't complain and might not find the new setting. Honestly, I can read white on green.
2. You update the color every 2500 milliseconds. Please use a listener so it updates faster.

If we could first implement another addon that may as well want to use the badge, we can flesh out the details like if all addons should have the same badge color, and how should we decide which get prio, etc.

**Comment by wgyt** [comment]
> This is ok as a temporary solution to just having one single addon using the badge. But I see 2 problems with this PR:
> 
> 1. The default color should be the current, just so current users don't complain and might not find the new setting. Honestly, I can read white on green.

I guess, but what is the default hex right now?

> 2. You update the color every 2500 milliseconds. Please use a listener so it updates faster.

Time to figure out what a listener is, while I'm in class :|

> If we could first implement another addon that may as well want to use the badge, we can flesh out the details like if all addons should have the same badge color, and how should we decide which get prio, etc.

I don't want to deal with this until #10 is done

**Comment by TheColaber** [comment]
`#8ce68c`, `addon.settings.addEventListener("change", callback)`

**Comment by wgyt** [comment]
> `#8ce68c`, `addon.settings.addEventListener("change", callback)`

Ok, all I need to do is figure out the callback, but I'll do that on my lunch break (1 hour from now)

**Comment by wgyt** [comment]
@TheColaber I added what you asked, could you test it?

File: addons/msg-count-badge/background.js [hunk 1]
```diff
@@ -5,5 +5,8 @@ export default async function ({ addon, global, console, setTimeout, setInterval
     addon.badge.color = addon.settings.get('color')
```
Comment (TheColaber) [inline addons/msg-count-badge/background.js:5]: remove this

Comment (wgyt) [reply addons/msg-count-badge/background.js:5]: Ok

Comment (wgyt) [reply addons/msg-count-badge/background.js:5]: Ok

**Comment by mxmou** [comment]
> The default color should be the current, just so current users don't complain and might not find the new setting. Honestly, I can read white on green.

You can, but not everyone can. Unlike Firefox, Chrome always uses white text on badges, and the current contrast doesn't meet any accessibility guidelines.

**Comment by wgyt** [comment]
> > The default color should be the current, just so current users don't complain and might not find the new setting. Honestly, I can read white on green.
> 
> You can, but not everyone can. Unlike Firefox, Chrome always uses white text on badges, and the current contrast doesn't meet any accessibility guidelines.

Can we just use a darkish blue?

**Comment by mxmou** [comment]
@wgyt I would use green because the badge is already green and most users wouldn't want a completely different color.

**Comment by wgyt** [comment]
Now that we have something that works, can we merge this?

**Comment by wgyt** [comment]
> > Now that we have something that works, can we merge this?
> 
> idk if format code with prettier is required but i think it is

it isn't. It fails ecause it doesn't have access to my fork

**Comment by WorldLanguages** [comment]
> Maybe `#008000` could be used instead, because it's already used to display the user's own username in comments.

That green is now the default. And users can rollback easily:

image

Any suggestions for one or two more presets? Some blue shade, maybe? Pure black?

**Comment by WorldLanguages** [comment]
@wgyt Whoops I can't push commits into this PR. Can you check "allow maintainers to idk what" on this PR?

**Comment by RedGuy12** [comment]
Maybe orange?
________________________________

**Pull request closed without merging**